src/core/init/duration_updater.ts

```diff
@@ -46,17 +46,32 @@ import { fromEvent } from "../../utils/event_emitter";
 const YEAR_IN_SECONDS = 365 * 24 * 3600;
 
 /**
- * Check if at least one the source buffer is currently updating.
+ * Emit when all the source buffers are not updating.
  * @param {SourceBufferList} sourceBuffers
- * @returns {Boolean}
+ * @returns {Observable}
  */
-function checkForUpdatingSourceBuffer(sourceBuffers: SourceBufferList): boolean {
+function sourceBuffersAreNotUpdating$(
+  sourceBuffers: SourceBufferList
+): Observable<undefined> {
+  const updatingSourceBuffers$: Array<Observable<boolean>> = [];
   for (let i = 0; i < sourceBuffers.length; i++) {
-    if (sourceBuffers[i].updating) {
-      return true;
-    }
+    const sourceBuffer = sourceBuffers[i];
+    updatingSourceBuffers$.push(
+      observableMerge(
+        observableFromEvent(sourceBuffer, "updatestart").pipe(mapTo(true)),
+        observableFromEvent(sourceBuffer, "updateend").pipe(mapTo(false))
```

Comment:
I think the event we want to listen to is juste "update" which indicates that that `updating` switched to `false` not "updateend" which is more about the task themselves (we don't care much about those, no)?

Also, as I said in the previous review (:P) we should also keep the interval because:
  - updating can change to `true` without an `"update"` event nor a `"updateend"` (e.g. `"error"` events)
  - we noticed an old bug with firefox where the event was not triggered